### ISSUE TITLE
Improved VoIP

### ISSUE THREAD

**Issue by nadonomy** [vector-im/roadmap#1]
__Summary__
VoIP in Element today is a second class citizen compared to sending and receiving messages. Several usability and stability issues exist throughout, and the experience for using native Matrix calls and Jitsi calls is inconsistent.

__Intended Outcome__
Unify the user experience between native Matrix calls & Jitsi calls so that it isn't jarring or confusing for users. Improve discoverability, usability and stability of calls all round.

**Comment by PxBCPG** [comment]
Integrated group calls would be awesome!

**Comment by Ezwen** [comment]
(I don't know if this is the right place for questions, do not hesitate to delete my message if it is not the case)

Will the new Voip implementation still rely on having a TURN server? Both for 1-1 calls and group calls?

**Comment by SimonBrandner** [comment]
> (I don't know if this is the right place for questions, do not hesitate to delete my message if it is not the case)
> 
> Will the new Voip implementation still rely on having a TURN server? Both for 1-1 calls and group calls?

Yes, as it will still use WebRTC

**Comment by networkException** [comment]
WebRTC does not necessarily need a TURN server

**Comment by SimonBrandner** [comment]
> WebRTC does not necessarily need a TURN server

No, it doesn't though most people will be behind NAT which makes use of a TURN server necessary, afaik. Maybe this can be avoided with an SFU but it can't be avoided for full-mesh calls

**Comment by SimonBrandner** [comment]
> > WebRTC does not necessarily need a TURN server
> 
> No, it doesn't though most people will be behind NAT which makes use of a TURN server necessary, afaik. Maybe this can be avoided with an SFU but it can't be avoided for full-mesh group calls

**Comment by PxBCPG** [comment]
Super cool!
And for when is it planned?

**Comment by dearjulian** [comment]
What do you mean by making calls easier to discover? 
Who should be able to discover what calls?

**Comment by SimonBrandner** [comment]
> What do you mean by making calls easier to discover? Who should be able to discover what calls?

One example would be that features shouldn't be hidden as screen-sharing used to be
